modules/puppet_proxy/customrun.rb

```diff
@@ -8,6 +8,6 @@ def run
       return false
     end
 
-    shell_command( [ escape_for_shell(cmd), Proxy::Puppet::Plugin.settings.customrun_args, shell_escaped_nodes ] )
+    shell_command( [ escape_for_shell(cmd), Proxy::Puppet::Plugin.settings.customrun_args || '', shell_escaped_nodes.join(' ') ] )
```

Comment:
It would be better to set a default value in https://github.com/theforeman/smart-proxy/blob/develop/modules/puppet_proxy/puppet_plugin.rb.